internal/service/wafv2/schemas.go

```diff
@@ -334,8 +334,51 @@ func fieldToMatchBaseSchema() *schema.Resource {
 		Schema: map[string]*schema.Schema{
 			"all_query_arguments": emptySchema(),
 			"body":                emptySchema(),
-			"method":              emptySchema(),
-			"query_string":        emptySchema(),
+			"cookies": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"match_scope": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice(wafv2.MapMatchScope_Values(), false),
+						},
+						"oversize_handling": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice(wafv2.OversizeHandling_Values(), false),
+						},
+						"match_pattern": {
+							Type:     schema.TypeList,
+							Required: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"all": {
```

Comment:
Hi @ewbankkit,
I seem to be facing an issue during the implementation of the CookieMatchPattern here.  'All' which is a struct without properties sounds rather like a boolean type, how do you suggest we  add this to the schema in the nicest way possible?

See here:
https://docs.aws.amazon.com/sdk-for-go/api/service/wafv2/#CookieMatchPattern

Reply:
@ewbankkit ☝️ thoughts?

Reply:
Finally getting to look at this...
@tiborhercz It looks like you have addressed this by using `emptySchema()`?

Reply:
Had a chat with @tiborhercz and adviced him to treat it as a struct as in the nearby future the api might add properties to this empty struct.